cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java

```diff
@@ -158,6 +158,10 @@ public void testMapreduceWithDynamicDatasets() throws Exception {
       .build());
     // build runtime args for app
     Map<String, String> runtimeArguments = Maps.newHashMap();
+
+    // Make sure there is only one mapper running at a time since this test has the Mapper writing
+    // to a dataset using increment and the in-memory table doesn't really support concurrent increment
```

Comment:
It should. However, the current dataset implementation structure such that this cannot be supported correctly, due to the `BufferingTable`. It is a known issue and is part of the dataset revamp effort than me and @anew is working on. In the meantime, have this workaround to not having the unit-test fail while still testing what this test was intended for (dynamic dataset access in MR).